crates/ra_assists/src/merge_match_arms.rs

```diff
@@ -0,0 +1,196 @@
+use hir::db::HirDatabase;
+use ra_syntax::{
+    ast::{MatchArm, AstNode,Pat,Whitespace,AstToken},
+    SyntaxElement,
+    Direction,
+    SyntaxKind::COMMA,
+    TextUnit,
+    algo::non_trivia_sibling
+};
+
+use crate::{AssistCtx, Assist, AssistId};
+
+pub(crate) fn merge_match_arms(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let match_arm = ctx.node_at_offset::<MatchArm>().unwrap();
+
+    let next_arm = MatchArm::cast(match_arm.syntax().next_sibling()?)?;
+
+    let match_arm_expr = match_arm.expr()?;
+
+    let next_arm_expr = next_arm.expr()?;
```

Comment:
These blank lines don't add much to readability imo.

Judicious use of blank lines indeed could be a **huge** improvement, because it allows to break code into logically related sections and bring focus to them. But adding too many blank lines devalues them: if everything focused, then nothing actually is.